pkg/cableengine/healthchecker/pinger.go

```diff
@@ -63,8 +63,16 @@ func (p *pingerInfo) sendPing() {
 
 	pinger.SetPrivileged(true)
 	pinger.RecordRtts = false
-	// After 3 seconds stop waiting.
-	pinger.Timeout = timeout
+
+	pinger.OnSend = func(packet *ping.Packet) {
+		// Pinger will be stopped once the packet loss reaches threshold and
+		// if packet loss is not 100% recreates the Pinger without marking an error
+		if pinger.Statistics().PacketsSent-pinger.Statistics().PacketsRecv > threshold {
+			klog.Infof("Packet loss reached a threshold of %d and hence stopping the pinger for HealthCheckIP: %q",
+				threshold, p.healthCheckIP)
```

Comment:
But, why would we stop it?.

I believe we should keep it up until connection is back, or the cable connection is removed.

Reply:
Done.